appinventor/blocklyeditor/src/blocks/lists.js

```diff
@@ -790,6 +790,61 @@ Blockly.Blocks['lists_minimum_number'] = {
   category : 'Lists',
   helpUrl : Blockly.Msg.LANG_LISTS_MIN_NUMBER_HELPURL,
   init : function() {
+      this.setColour(Blockly.LIST_CATEGORY_HUE);
+      this.appendValueInput('LIST')
+        .setCheck(Blockly.Blocks.Utilities.YailTypeToBlocklyType("list",Blockly.Blocks.Utilities.INPUT))
+        .appendField(Blockly.Msg.LANG_LISTS_MIN_NUMBER_INPUT_MIN, 'TITLE')
+        .setAlign(Blockly.ALIGN_RIGHT);
+      this.appendDummyInput('DESCRIPTION')
+        .appendField(Blockly.Msg.LANG_LISTS_MIN_NUMBER_INPUT_COMPARATOR)
+        .appendField(new Blockly.FieldParameterFlydown(Blockly.Msg.LANG_LISTS_MIN_NUMBER_INPUT_VAR1,
+                                                       true, // name is editable
+                                                       Blockly.FieldFlydown.DISPLAY_BELOW),
+                     'VAR1')
+        .appendField(Blockly.Msg.LANG_LISTS_MIN_NUMBER_INPUT_AND)
+        .appendField(new Blockly.FieldParameterFlydown(Blockly.Msg.LANG_LISTS_MIN_NUMBER_INPUT_VAR2,
+                                                       true, // name is editable
+                                                       Blockly.FieldFlydown.DISPLAY_BELOW),
+                     'VAR2')
+        .appendField(Blockly.Msg.LANG_LISTS_MIN_NUMBER_INPUT_COMPARATOR2)
+        .setAlign(Blockly.ALIGN_RIGHT);
+      this.appendIndentedValueInput('COMPARE');
+      this.setOutput(true, null);
+      this.setTooltip(Blockly.Msg.LANG_LISTS_MIN_NUMBER_TOOLTIP);
+      this.setInputsInline(false);
+    },
+    saveConnections: Blockly.saveConnections,
+    getVars: function() {
+      return [this.getFieldValue('VAR')];
+    },
+    blocksInScope: function() {
+      var compareBlock = this.getInputTargetBlock('COMPARE');
+      if (compareBlock) {
+        return [compareBlock];
+      } else {
+        return [];
+      }
+    },
+    declaredNames: function() {
+      return [this.getFieldValue('VAR')];
+    },
+    renameVar: function(oldName, newName) {
+      if (Blockly.Names.equals(oldName, this.getFieldValue('VAR'))) {
+        this.setFieldValue(newName, 'VAR');
+      }
+    },
+    freeVariables: function() { // return the free lexical variables of this block
+      // Should return the empty set: something is wrong if it doesn't!
+      var result = Blockly.LexicalVariable.freeVariables(this.getInputTargetBlock(this.bodyInputName));
+      result.subtract(new Blockly.NameSet(this.declaredNames()));
+      if (result.isEmpty()) {
+        return result;
+      } else {
+        throw "Violation of invariant: procedure declaration has nonempty free variables: " + result.toString();
```

Comment:
This error message should be wrapped in a call to `Error()`.

Reply:
So would it be Error("Violation of invariant: procedure declaration has nonempty free variables: " + result.toString());? Cause this is how it is implemented with other blocks